catalyst/src/gemini.test.js

```diff
@@ -49,52 +49,141 @@ describe('validate_questionnaire_format', () => {
     };
     expect(validate_questionnaire_format(questionnaire)).toBe(false);
   });
+
+  test('rejects mixed types in values', () => {
```

Comment:
Consider adding a brief comment in the code explaining why mixed types are invalid, for future reference.